website/statistics.py

```diff
@@ -362,3 +515,101 @@ def get_general_class_stats(students):
         "week": {"runs": weekly_successes + weekly_errors, "fails": weekly_errors},
         "total": {"runs": successes + errors, "fails": errors},
     }
+
+
+# append the programs obtained on each level
+def find_program_runs_per_level(program_runs_per_level, success_runs_per_level, error_runs_per_level, programs, level):
+    program_runs_per_level.append([])
+    success_runs_per_level.append([]), error_runs_per_level.append([])
+    for program in programs:
+        if program['level'] == level:
+            program_runs_per_level[level - 1].append([program['name'], str(program.get("error"))])
+            if str(program.get("error")) == "True":
+                success_runs_per_level[level - 1].append(program['name'])
+            elif str(program.get("error")) == "False":
+                error_runs_per_level[level - 1].append(program['name'])
+
+
+# append the quizzes obtained on each level
+def find_quizzes_per_level(quizzes_per_level, quizzes, level):
+    quizzes_per_level.append([])
+    for quiz_score in quizzes:
+        if quiz_score['level'] == level:
+            quizzes_per_level[level - 1].append(quiz_score["scores"][0])
+
+
+def calc_avg_quizzes_per_level(avg_quizzes_ran_per_level, quizzes_ran_per_level, level):
+    if not quizzes_ran_per_level[level - 1]:
+        avg_quizzes_ran_per_level.append(0)
+    else:
+        quizzes_per_level = quizzes_ran_per_level[level - 1]
+        if len(quizzes_ran_per_level[level - 1]) == 1:
+            avg_quizzes_ran_per_level.append(quizzes_per_level[0])
+        if len(quizzes_ran_per_level[level - 1]) > 1:
+            average_quiz_score_per_level = 0
+            for index in range(1, len(quizzes_per_level) + 1):
+                average_quiz_score_per_level += quizzes_per_level[index - 1]
+
+            average_quiz_score_per_level /= (len(quizzes_ran_per_level[level - 1]))
+            avg_quizzes_ran_per_level.append(int(average_quiz_score_per_level))
+
+    return avg_quizzes_ran_per_level
```

Comment:
Please review this code, because I tried running some quiz attemps and it only computes one attempt. Also, for taking the average you can do something like `sum(quizzes_per_level)/len(quizzes_per_level)`

Reply:
Issue fixed, it seems that Hedy database saves quizzes that are completed in the same week differently than if they are done in different weeks.